src/Map.tsx

```diff
@@ -34,6 +34,13 @@ const Map: React.FC<MapProps> = ({
     zoom: 16
   });
 
+  const [hoverSource, setHoverSource] = useState('');
+
+  const handleHover = useCallback(({ features }: MapEvent) => {
+    setHoverSource(features && features.length > 0 ? features[0].source : '');
```

Comment:
Maybe it'd me more readable to replace `features &&` with `Array.isArray(features) &&`

Reply:
Would it be possible to make the click a factor?
I think it'd look a little cooler if the 'grab' cursor only appeared only after I held a click w/ my mouse